mne/decoding/gsoc.py

```diff
@@ -0,0 +1,79 @@
+# Authors: Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from .mixin import TransformerMixin
+from ..epochs import _BaseEpochs
+
+
+class EpochsTransformerMixin(TransformerMixin):
+    def __init__(self, n_chan=None):
+        self.n_chan = n_chan
+        self._check_init()
+
+    def fit(self, X, y=None):
+        return self
+
+    def fit_transform(self, X, y=None):
+        self.fit(X, y)
+        return self.transform(X)
+
+    def _reshape(self, X):
+        """Recontruct epochs to get a n_trials * n_chan * n_time"""
+        if isinstance(X, _BaseEpochs):
+            X = X.get_data()
+            # TODO: pick data channels (EEG/MEG/SEEG/ECOG if epochs)
+        elif not isinstance(X, np.ndarray):
+            raise ValueError('X must be an Epochs or a 2D or 3D array, got '
+                             '%s instead' % type(X))
+        elif (X.ndim != 3) or (self.n_chan is None):
+            n_epoch = len(X)
+            n_time = np.prod(X.shape[1:]) // self.n_chan
```

Comment:
Alright. So we keep it the same if its 3D. I think the condition above should be `if self.n_chan is not None`. 
```

make_pipeline(
    Resampler(n_chan=64),  # Take 2D X, return 3D
    TimeFrequencyDecomposer(),  # Take 3D  X return 4 D
    Connectivity(),
    FooBar(),
    ...
    Vectorizer(),  # my transformers output X with more than 2D, wheras the following sklearn steps expect 2D X
    PCA(),
    LogisticRegression()
)
```
Citing the example in from dropbox paper.
Following `TimeFrequencyDecomposer` if it takes, say a 3D input, how will it convert it from 4D? Isn't `n_chan` required then? Or is it always of shape `n_trials, n_chan, ....` ?